### PR TITLE
SCB-1580 Configurable oas style check rules

### PR DESCRIPTION
**Pull request by chanjarster**
1. Make *ValidatorFactory create *Validators by FactoryOptions
1. Remove @Configuration and use *ValidatorFactory instead
1. Update cli and oas-validator-web to adapt to the change mentioned above
1. Some unit test codes

### CODE REVIEW

File: oas-validator/README-ZH.md [hunk 1]
```diff
@@ -8,154 +8,139 @@ OpenAPI V3 Spec校验工具。
 * oas-validator-core-spring，骨架的Spring Boot Starter
 * oas-validator-test，核心API的测试帮助类
 * oas-validator-compliance，合规性校验实现
-* oas-validator-compliance-spring，合规性校验的Spring Boot Starter
```
Comment (WillemJiang) [inline oas-validator/README-ZH.md]: why did you remove this line.

Comment (chanjarster) [reply oas-validator/README-ZH.md]: oas-validator-compliance-spring directly @Configure *Validator beans, which are not configurable. While we use *ValidatorFactory now, so we can abandon it.

Related commit:
* https://github.com/apache/servicecomb-toolkit/pull/52/commits/ece6c372630deb4b24ddbfb64fd389eba4cb3572
* https://github.com/apache/servicecomb-toolkit/pull/52/commits/9c9c22f03e20362b976e067c3d67e589988db727

**Comment by chanjarster** [comment]
@WillemJiang what this PR does are:

1. Make *ValidatorFactory create *Validators by FactoryOptions
2. Remove @Configuration and use *ValidatorFactory instead
3. Update cli and oas-validator-web to adapt to the change mentioned above
4. Some unit test codes
